### PR TITLE
[WIP] Allow to disable a user.

### LINKED ISSUE #4237: Unable to disable users
The `Bolt\Storage\Entity\Users` entity will return the default property of the class for the `$enabled` property… Flipping the value has the opposite effect.

**Note to self:** When this is fixed, re-enable this test: https://github.com/bolt/bolt/commit/cba2038734154a9e8ee3520b7daf084cfedd7114

### PR DESCRIPTION
**Pull request by wysow**
The default value will be set at database level when creating a user so this fix is pretty straightforward but not sure it's the perfect way to go, what do you think?

Fixes #4237

### CODE REVIEW

**Comment by GawainLynch** [comment]
Oh you may have bitten off a lot here… :grin: 

See #4237  as there is something funky going on in the storage layer with this one.

**Comment by wysow** [comment]
@GawainLynch not sure to get you on this one... My `manual` tests are going ok, waiting for the Travis build to see what I missed...

**Comment by wysow** [comment]
@GawainLynch where can we talk more generously about this?

**Comment by GawainLynch** [comment]
#boltcms on Freenode is best

**Comment by GawainLynch** [comment]
This is what you get on First User with these changes:
first-user

File: src/EventListener/StorageEventListener.php [hunk 1]
```diff
@@ -86,6 +88,20 @@ protected function passwordHash(Entity\Users $usersEntity)
...
+    /**
+     * Set user as enabled if user is new
+     *
+     * @param Entity\Users $usersEntity
+     */
+    protected function enableUser(Entity\Users $usersEntity)
+    {
+        if ($usersEntity->getShadowSave()) {
+            return;
+        } elseif (null === $usersEntity->getId()) {
```
Comment (GawainLynch) [inline src/EventListener/StorageEventListener.php]: Before the coding :cop: gets here :police_car: … Yoda conditions avoid we do.

**Comment by GawainLynch** [comment]
…and then there is the embarrassing moment with the :koala: [leaves this in by accident](https://github.com/bolt/bolt/blob/master/src/AccessControl/Login.php#L150).

File: src/EventListener/StorageEventListener.php (hunk 1 above)
Comment (wysow) [inline src/EventListener/StorageEventListener.php]: Does it mean that you don't like this implementation?

Comment (GawainLynch) [inline src/EventListener/StorageEventListener.php]: The standards we use are:
```php
        } elseif ($usersEntity->getId() === null) {
```

**Comment by wysow** [comment]
@GawainLynch I think I'm too new on Bolt to understand this Travis build error, do you have an hint?

**Comment by GawainLynch** [comment]
Unsure… But I can look in the morning for you… Wonderful effort to get this far! :+1:

**Comment by wysow** [comment]
Thanks, will do the same on my own, let's get posted.

**Comment by GawainLynch** [comment]
The first one needs this:
```diff
diff --git a/tests/phpunit/unit/AccessControl/LoginTest.php b/tests/phpunit/unit/AccessControl/LoginTest.php
index b80b69d..dd6abc8 100644
--- a/tests/phpunit/unit/AccessControl/LoginTest.php
+++ b/tests/phpunit/unit/AccessControl/LoginTest.php
@@ -57,9 +57,9 @@ class LoginTest extends BoltUnitTest
         $app = $this->getApp();
         $this->addDefaultUser($app);
 
-        $logger = $this->getMock('\Monolog\Logger', ['error'], ['testlogger']);
+        $logger = $this->getMock('\Monolog\Logger', ['alert'], ['testlogger']);
         $logger->expects($this->atLeastOnce())
-            ->method('error')
+            ->method('alert')
             ->with($this->equalTo("Attempt to login with disabled account by 'admin'"));
         $app['logger.system'] = $logger;
 
```

**Comment by GawainLynch** [comment]
This is the second one:
```diff
diff --git a/tests/phpunit/unit/Controller/Backend/AuthenticationTest.php b/tests/phpunit/unit/Controller/Backend/AuthenticationTest.php
index ce02d2f..72d6aff 100644
--- a/tests/phpunit/unit/Controller/Backend/AuthenticationTest.php
+++ b/tests/phpunit/unit/Controller/Backend/AuthenticationTest.php
@@ -92,10 +92,10 @@ class AuthenticationTest extends ControllerUnitTest
     {
         $app = $this->getApp();
         $loginMock = $this->getLoginMock($app);
-        $loginMock->expects($this->any())
+        $loginMock->expects($this->once())
             ->method('login')
             ->will($this->returnValue(true));
-        $this->setService('authentication.login', $loginMock);
+        $this->setService('access_control.login', $loginMock);
 
         $this->setSessionUser(new Entity\Users(['username' => 'test', 'roles' => []]));
 
```

**Comment by wysow** [comment]
@GawainLynch Thx a lot it's done let see if it's green now...

**Comment by GawainLynch** [comment]
Passes locally and Travis has gone green for PHP 7 :grin:

**Comment by wysow** [comment]
@GawainLynch Should I update this: https://scrutinizer-ci.com/g/bolt/bolt/inspections/a02bd9e8-dcee-4008-b9dd-770fb59f59f3/issues/files/src/EventListener/StorageEventListener.php?status=new&orderField=path&order=asc ?

**Comment by GawainLynch** [comment]
Sure, if you want.

**Comment by GawainLynch** [comment]
:shipit:

**Pull request merged**